internal/servers/rtsp/session.go

```diff
@@ -269,6 +270,16 @@ func (s *session) onPlay(_ *gortsplib.ServerHandlerOnPlayCtx) (*base.Response, e
 		s.state = gortsplib.ServerSessionStatePlay
 		s.transport = s.rsession.SetuppedTransport()
 		s.mutex.Unlock()
+
+		for _, u := range s.stream.CachedUnits {
+			for _, pkt := range u.GetRTPPackets() {
+				for _, medi := range s.stream.Desc().Medias {
+					if medi.Type == description.MediaTypeVideo {
+						s.rsession.WritePacketRTP(medi, pkt)
```

Comment:
Should this be done when the mutex is acquired above?

Reply:
It does not seems to be required.

Reply:
Why isn't it required?  Are we not worried the thread writing to CachedUnits might modify or truncate CachedUnits while we're in the middle of this loop?

Could we make a copy of CachedUnits inside the mutex then use it outside the mutex?  Just a shallow copy would be sufficient, since the actual cached units are immutable.

Reply:
This is the RTSP path, which isn't used in the WebRTC playback. I do want to get this right as it would impact other use cases.